Support for GHC 9.10

wz1000 opened the pull request:
Needs a ghc-exactprint release, currently builds against https://github.com/alanz/ghc-exactprint/tree/ghc-9.10

wz1000 committed:
9.10 support

wz1000 committed:
exactprint cabal compat

wz1000 commented:
@pepeiborra this should be ready to merge.

watashi commented:
Can we also add ghc 9.10.1 to CI? or ghc 9.10 support can be broken again without notice in the future. 

https://github.com/facebookincubator/retrie/blob/main/.github/workflows/test.yml#L12,L16

pepeiborra commented:
Ping @wz1000

wz1000 committed:
Update CI to build with 9.10

wz1000 commented:
I am looking at the test failures.

wz1000 committed:
attempt some testsuite fixes

alanz committed:
Clean up a bit

- Remove earlier comments from previous upgrade
- Add CPP to silence a bunch of warnings

alanz committed:
WIP. drop from 32 to 28 failures

alanz committed:
Fix backticks test

By adjusting the paren addition code, 9.10 exactprint uses spans for
delta calcs, we can keep them

alanz committed:
Some cleanups

alanz committed:
Fix spacing for fold/unfold function rules

Down to 19 failures

alanz committed:
Fix parsing adhoc rules

Drops to 17/86

alanz committed:
Non-functional changes

whitespace, commenting debug bits

alanz committed:
Drop test test failures to 13

alanz committed:
Fix adding imports

now 13/86

alanz committed:
Fix test /Adhoc/

12/86

alanz committed:
Fix golden.Imports

11/86

alanz committed:
Fix Types2.test, and template parsing

When we finalise the templates, do makeDeltaAst for all variants.

This fixes Types2 and Parens tests, but breaks

Readme1
Types3
Types4

alanz committed:
Only makeDeltaAst for the rule templates

We need the patterns to be located, to not replace the original when
matching.

Down to 6/86

alanz committed:
Fix up fixity spans in fixOnePat

Brings it to 5/86 remaining failures

alanz committed:
Explicitly store the original pattern location in Query

This is used to filter out potential matches in the file the query
originates in.

But often we have had to re-arrange it with deltas at the point of
matching, so can no longer easily extract it from the pattern to be
matched.

Instead we store it explicitly when making a Query

alanz committed:
Use originating location for pattern

Still 5/86 failures, but fewer failing instances in PatternSyn2

alanz committed:
Keep the original anchor when replacing

This sorts out surrounding spacing.

Down to 4/86

alanz committed:
Preserve var locations when converting a pat to an expr

Down to 3/86

alanz committed:
Apply fixities correctly

Recursion /is/ needed, because we are moving things around at this
level, the child nodes may need re-arranging as a result

Still 3/86, but they all relate to comments. I think.

alanz committed:
Add compatibility layer for ghc-exactprint 1.10

Which has made most exactprint transform functions pure, rather than
in TransFormT.

alanz committed:
WIP
